### PR TITLE
Us 20

### LINKED ISSUE #49: Promoting pawns - US_20
Description: As user I want to be able to promote pawns according to chess rules.
Acceptance criteria: When promoting a pawn in international chess there is a visual option to choose any eligible piece and when selecting one of these the pawn is replaced by that piece.

### PR DESCRIPTION
**Pull request by MartinRuz**
# Purpose
closes #49 

# known bugs
-changing window size while you have a dropdown menu for promotion will cause the placement to be wrong, this is a minor issue and will maybe be fixed later

# todo later
-when US6 is merged I will make the piecestyle consistent

# Pre-merge TODOs and Checks 
- [x] _(exclude checks that are not relevant for your feature)_
- [x] PGN file loads correctly
- [x] Navigating through the move history works
- [x] Starting the engine shows moves (test for both sides)
- [x] Selecting a different variant loads a new board with the variant and you can make moves on the board
- [x] In Crazyhouse the pocket pieces are shown and you can drop pieces

### CODE REVIEW

File: src/renderer/components/PromotionModal.vue [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    return {
+      promDir: { 1: 'q', 2: 'n', 3: 'r', 4: 'b' }
+    }
+  },
+  computed: {
+    pieceStyle () {
+      return this.$store.getters.pieceStyle
+    },
+    turn () {
+      if (this.$store.getters.turn === undefined) {
```
Comment (ptrmdr) [inline src/renderer/components/PromotionModal.vue:45]: I think undefined would be considered false in a condition so this check is not really necessary.

File: src/renderer/components/ChessGround.vue [hunk 2]
```diff
@@ -216,6 +229,42 @@ export default {
     })
   },
   methods: {
+    showPromotionModal () {
```
Comment (ptrmdr) [inline src/renderer/components/ChessGround.vue:232]: This code is not very clear to me. Maybe there is a more elegant way (with vuejs?).

**Review by ptrmdr — COMMENTED** [review]
This seems to work.

Please remove unnecessary/commented out code (e.g. some of the console.log()) and also have a look at my comments.

**Comment by BododB** [comment]
Currently when the promotion box is displayed and a move from the history is selected, the promotion box remains open.

**Comment by Zerthox** [comment]
Promotion functionality looks good. When testing I found the UI a bit unintuitive. Maybe we can add a border or box shadow to separate the modal from the board? Also it would be nice if there is an on hover effect for the pieces in the promotion modal.

**Pull request merged**